pre_commit/languages/download.py

```diff
@@ -0,0 +1,291 @@
+        hasher = hashlib.new(self.algorithm)
+        while buffer := io.read(chunk):
+            hasher.update(buffer)
+            yield buffer
+        digest = b64encode(hasher.digest()).decode('utf-8)')
+        if digest != self.checksum:
+            raise ChecksumMismatchError(
+                expected=self.checksum,
+                actual=digest,
+            )
+
+
+@dataclass(frozen=True)
+class URI:
+    value: str
+
+    def __post_init__(self) -> None:
+        url = urlparse(self.value)
+        if not all([url.scheme, url.netloc]):
+            raise ValueError(f'Invalid URI: {self.value}')
+
+    def __str__(self) -> str:
+        return self.value
+
+
+@dataclass(frozen=True)
+class Metadata:
+    value: str
+
+    @property
+    def parts(self) -> tuple[str, str, str, str]:
+        first, second, third, fourth = self.value.splitlines()
+        return (first, second, third, fourth)
+
+    @property
+    def platform(self) -> Platform:
+        platform, _, _, _ = self.parts
+        return Platform(platform)
+
+    @property
+    def sri(self) -> SRI:
+        _, sri, _, _ = self.parts
+        return SRI(sri)
+
+    @property
+    def uri(self) -> URI:
+        _, _, uri, _ = self.parts
+        return URI(uri)
+
+    @property
+    def filename(self) -> PurePath:
+        _, _, _, path = self.parts
+        return PurePath(path)
+
+
+def download(uri: URI, sri: SRI, filename: Path) -> None:
+    with urlopen(str(uri)) as ws:
+        with filename.open('wb') as fp:
+            for buffer in sri.check(ws):
+                fp.write(buffer)
+            fp.flush()
+    chmod(filename, stat.S_IRUSR | stat.S_IXUSR)
+
+
+def install_environment(
+        prefix: Prefix,
+        version: str,
+        additional_dependencies: Sequence[str],
+) -> None:
+    host = Platform.host()
+    for dep in additional_dependencies:
+        m = Metadata(dep)
+        if host == m.platform:
+            envdir = Path(
+                lang_base.environment_dir(
+                    prefix, ENVIRONMENT_DIR,
+                    version,
+                ),
+            )
+            envdir.mkdir(parents=True, exist_ok=True)
+            filename = Path(m.filename)
+            download(m.uri, m.sri, envdir / filename)
+            srisum = envdir / 'health.srisum'
+            with srisum.open('w', encoding='utf8') as stream:
+                stream.write(f'{m.sri}  {filename}\n')
+            return
+    raise KeyError(f'Failed to find platform `{host}` in \
+`additional_dependencies`: {additional_dependencies}')
+
+
+def health_check(prefix: Prefix, version: str) -> str | None:
+    envdir = Path(lang_base.environment_dir(prefix, ENVIRONMENT_DIR, version))
+    srisum = envdir / 'health.srisum'
+    with srisum.open(encoding='utf8') as stream:
+        for line in stream:
+            sri_str, filepath = line.strip().split(' ', 1)
+            sri_str = sri_str.strip()
+            filepath = filepath.strip()
```

Comment:
Splitting on a double space should be all that you need, why are you stripping the parts, it should be unnecessary.

Whilst it is a complete edge case, this would prevent filesystem paths with whitespace at the start/end.

```suggestion
            integrity, filepath = line.strip().split('  ', 1)
```

Reply:
Done